gnucash/gnome-utils/gnc-embedded-window.c

```diff
@@ -233,20 +193,18 @@ gnc_embedded_window_class_init (GncEmbeddedWindowClass *klass)
  *  @param klass A pointer to the class data structure for this
  *  object. */
 static void
-gnc_embedded_window_init (GncEmbeddedWindow *window,
-                          GncEmbeddedWindowClass *klass)
+gnc_embedded_window_init (GncEmbeddedWindow *window)
 {
     ENTER("window %p", window);
 
     gtk_orientable_set_orientation (GTK_ORIENTABLE(window), GTK_ORIENTATION_VERTICAL);
 
     // Set the style context for this widget so it can be easily manipulated with css
-    gnc_widget_set_style_context (GTK_WIDGET(window), "GncEmbededWindow");
+    gnc_widget_set_style_context (GTK_WIDGET(window), "GncEmbeddedWindow");
 
     gnc_embedded_window_setup_window (window);
 
-    gnc_gobject_tracking_remember(G_OBJECT(window),
-                                  G_OBJECT_CLASS(klass));
+    gnc_gobject_tracking_remember(G_OBJECT(window), NULL);
     LEAVE(" ");
```

Comment:
The answer on the G_DEFINE_TYPE bug](https://gitlab.gnome.org/GNOME/glib/issues/1621) was "too bad, just unroll the macro", which suggests an alternative: Copy the  G_DEFINE_TYPE.* macros that we need to GNC_DEFINE_TYPE.* and modify GNC_DEFINE_TYPE to use a correct GInstanceInitFunc.

Reply:
That sounds like a possible way to go. I'll try it. Thank's for doing the request to the glib people.

Reply:
I'll put correcting the name of the css class into a separate pull request.

Reply:
John, would you redefine the macro or define similar ones that provide an init function with two parameters ? And where would you put these definitions. They should surely go to a central location. And should we mix the ones with one parameter and that with two depending on the use case ? I suppose in the current use case it would affect only G_DEFINE_TYPE_WITH_CODE which could be changed to G_DEFINE_TYPE_WITH_CODE_INIT_TWO_PARAMS.

Reply:
Neither. You want to create **GNC**_DEFINE_TYPE_WITH_CODE that calls _**GNC**_DEFINE_TYPE_EXTENDED_BEGIN. The latter will be a copy of [_G_DEFINE_TYPE_EXTENDED_BEGIN](https://gitlab.gnome.org/GNOME/glib/blob/master/gobject/gtype.h#L1949) except that the first line will be
```C
static void     type_name##_init              (TypeName        *self, void *class); \
```
so that it will work with classes having two-parameter init functions.
I suppose it should go in gnucash/gnome-utils/gnc-gobject-utils.h.

Reply:
ok. Sounds good.